DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -151,13 +94,10 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
                 throw new DeviceRequiredPropertyNotFoundException("'DeviceID' property is missing");
             }
 
-            string deviceId = device.DeviceProperties.DeviceID;
-
-            DeviceModel existingDevice = await this.GetDeviceAsync(deviceId);
-
+            DeviceModel existingDevice = await GetDeviceAsync(device.id);
```

Comment:
@handeadiguzel Yup you're right. One spot was using GetDeviceAsync(device.id) instead of GetDeviceAsync(device.DeviceProperties.DeviceID). Fixing